sycl/test/basic_tests/aspect_usage.cpp

```diff
@@ -0,0 +1,50 @@
+// Test checks Propagate Aspect Usage pass.
+
+// RUN: %clangxx -fsycl -fno-legacy-pass-manager %s -emit-llvm -S -o - | FileCheck %s --check-prefix CHECK-NODES
+// RUN: %clangxx -fsycl -fno-legacy-pass-manager -fno-sycl-early-optimizations %s -emit-llvm -S -o - | FileCheck %s --check-prefix CHECK-NODES
+// RUN: %clangxx -fsycl -flegacy-pass-manager %s -emit-llvm -S -o - | FileCheck %s --check-prefix CHECK-NODES
+// RUN: %clangxx -fsycl -flegacy-pass-manager -fno-sycl-early-optimizations %s -emit-llvm -S -o - | FileCheck %s --check-prefix CHECK-NODES
+
+// fp16 aspect corresponds to constant 5.
+// CHECK-NODES-DAG: !intel_used_aspects ![[NODE:[0-9]+]]
+// CHECK-NODES: ![[NODE]] = !{i32 5}
+
+// RUN: %clangxx -fsycl -fno-legacy-pass-manager %s -emit-llvm -S -o /dev/null 2>&1 | FileCheck %s --check-prefix CHECK-WARNINGS
+// RUN: %clangxx -fsycl -flegacy-pass-manager %s -emit-llvm -S -o /dev/null 2>&1 | FileCheck %s --check-prefix CHECK-WARNINGS
+
+// RUN: %clangxx -fsycl -fno-legacy-pass-manager %s -g -emit-llvm -S -o /dev/null 2>&1 | FileCheck %s --check-prefix CHECK-WARNINGS-DBG
+// RUN: %clangxx -fsycl -flegacy-pass-manager %s -g -emit-llvm -S -o /dev/null 2>&1 | FileCheck %s --check-prefix CHECK-WARNINGS-DBG
+
+// CHECK-WARNINGS: warning: function '_Z5func1v' uses aspect '5' not listed in `sycl::requires()`
+// CHECK-WARNINGS-NEXT:  _Z5func1v()
+// CHECK-WARNINGS-NEXT:  _Z5func2v()
+// CHECK-WARNINGS-NEXT:  _Z5func3v()
+// CHECK-WARNINGS-NEXT: compile with '-g' to get source location
+
+// CHECK-WARNINGS-DBG: warning: function '_Z5func1v' uses aspect '5' not listed in `sycl::requires()`
```

Comment:
Could you please clarify why warning message says "sycl::requires()"? Shouldn't the aspect be listed in "sycl::device_has()"? Is "sycl::requires()" is the same as "sycl::device_has()"?

Reply:
Yes, you're right. I copied it from [design doc](https://github.com/gmlueck/llvm/blob/1e927c107a7ac5c486a5620fba4ce80912701785/sycl/doc/OptionalDeviceFeatures.md#new-llvm-ir-pass-to-propagate-aspect-usage). In fact, it should be `sycl::device_has()`